apis/python/src/tiledb/vector_search/vamana_index.py

```diff
@@ -111,11 +148,135 @@ def query_internal(
         queries = np.transpose(queries)
         if not queries.flags.f_contiguous:
             queries = queries.copy(order="F")
-        queries_feature_vector_array = vspy.FeatureVectorArray(queries)
+        if mode is None:
+            queries_feature_vector_array = vspy.FeatureVectorArray(queries)
+            distances, ids = self.index.query(queries_feature_vector_array, k, opt_l)
+            return np.array(distances, copy=False), np.array(ids, copy=False)
+        else:
+            return self.taskgraph_query(
+                queries=queries,
+                k=k,
+                opt_l=opt_l,
+                mode=mode,
+                resource_class=resource_class,
+                resources=resources,
+                num_workers=num_workers,
+                config=self.config,
+            )
+
+    def taskgraph_query(
+        self,
+        queries: np.ndarray,
+        k: int,
+        opt_l: int,
+        mode: Mode,
+        resource_class: Optional[str],
+        resources: Optional[Mapping[str, Any]],
+        num_workers: int,
+        config: Optional[Mapping[str, Any]],
+    ):
+        """
+        Query an VAMANA index using TileDB cloud taskgraphs
+
+        Parameters
+        ----------
+        queries: numpy.ndarray
+            ND Array of queries
+        k: int
+            Number of top results to return per query
+        opt_l: int
+            How deep to search. Should be >= k.
+        mode: Mode
+            If provided the query will be executed using TileDB cloud taskgraphs.
+            For distributed execution you can use REALTIME or BATCH mode.
+            For local execution you can use LOCAL mode.
+        resource_class:
+            The name of the resource class to use ("standard" or "large"). Resource classes define maximum
+            limits for cpu and memory usage. Can only be used in REALTIME or BATCH mode.
+            Cannot be used alongside resources.
+            In REALTIME or BATCH mode if neither resource_class nor resources are provided,
+            we default to the "large" resource class.
+        resources:
+            A specification for the amount of resources to use when executing using TileDB cloud
+            taskgraphs, of the form: {"cpu": "6", "memory": "12Gi", "gpu": 1}. Can only be used
+            in BATCH mode. Cannot be used alongside resource_class.
+        num_workers: int
```

Comment:
This is not performing distributed computation with many workers. It is only running the centralized query execution using cloud and not client resources.

Lets not have placeholder args and documentation when the implementation is not yet in place.

Reply:
Ah, yes, accident to include this, will remove.